app/src/main/java/com/android/voyageur/ui/trip/activities/ActivitiesForOneDay.kt

```diff
@@ -46,36 +55,81 @@ fun ActivitiesForOneDayScreen(
       tripsViewModel.selectedDay.value
           ?: return Text(text = "No day selected. Should not happen", color = Color.Red)
 
+  // States for filtering and search
+  var selectedFilters by remember { mutableStateOf(setOf<ActivityType>()) }
+  var searchQuery by remember { mutableStateOf("") }
+
   var activities by remember {
     mutableStateOf(
         tripsViewModel
             .getActivitiesForSelectedTrip()
             .filter {
-              it.startTime.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate() == day
+              val matchesDate =
+                  it.startTime.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate() ==
+                      day
+              val matchesSearch =
+                  searchQuery.isEmpty() || it.title.contains(searchQuery, ignoreCase = true)
+              val matchesFilter = selectedFilters.isEmpty() || it.activityType in selectedFilters
+              matchesDate && matchesSearch && matchesFilter
+            }
+            .sortedWith(compareBy({ it.startTime }, { it.endTime })))
+  }
```

Comment:
Actually, this entire block of code is duplicated in two places in the new changes. It would be beneficial to refactor this using a separate function such that the code is more maintainable and readable.